plugin/keyring_vault/vault_memory.h

```diff
@@ -0,0 +1,47 @@
+#ifndef MYSQL_VAULT_MEMORY
+#define MYSQL_VAULT_MEMORY
+
+#include <my_global.h>
+#include <string.h>
+#include "keyring_memory.h"
+
+namespace keyring
+{
+#if !defined(HAVE_STDCXX11) && !defined(WIN32)
+  inline void memset_s(void *dest, size_t dest_max, int c, size_t n)
+  {
+    volatile unsigned char *p = reinterpret_cast<unsigned char*>(dest);
+    while (dest_max-- && n--) {
+      *p++ = c;
+    }
+  } 
+#endif
+
+  template <class T> class Secure_allocator : public std::allocator<T>
```

Comment:
A class should not be inherited if it has not virtual destructor - provided it can be delete by pointer to a base class, which is not true in case of allocator for basic_string. Please see http://stackoverflow.com/questions/21081796/why-not-to-inherit-from-stdallocator . Is there any other consideration it should not be a public inheritance and private one ?